src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression
+{
+    using static SyntaxFactory;
+
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal sealed class ConvertSwitchStatementToExpressionCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(IDEDiagnosticIds.ConvertSwitchStatementToExpressionDiagnosticId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(
+                new MyCodeAction(c => FixAsync(context.Document, context.Diagnostics.First(), c)),
+                context.Diagnostics);
+            return Task.CompletedTask;
+        }
+
+        protected override Task FixAllAsync(Document document, ImmutableArray<Diagnostic> diagnostics, SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var root = editor.OriginalRoot;
+            foreach (var diagnostic in diagnostics)
+            {
+                var switchStatement0 = (SwitchStatementSyntax)root.FindNode(diagnostic.Location.SourceSpan);
```

Comment:
strang use of 0.

Reply:
also, shoudl likely not be using the diag location. otherwise, thsi means that the entire switch would be squiggled.  better to use .AdditionalLocations for this.

Reply:
cast seems unnecessary.

Reply:
"strang use of 0." because C# doesn't allow variable shadowing. This is a conventional in the branch.

Reply:
but you don't reference this below... so why does this need a name, let alone a name that could collide.  just call it 'node' or something :)